agent-install/edgeNodeFiles.sh

```diff
@@ -511,7 +511,7 @@ all_main() {
 
     cloudLogin
 
-    if [[ -n $DIR ]]; then pushd; fi   # if they want the files somewhere else, make that our current dir
+    if [[ -n $DIR && -d $DIR ]]; then pushd $DIR; fi   # if they want the files somewhere else, make that our current dir
```

Comment:
This isn't quite right. The effect of `&& -d $DIR` is that if the user specified `-f <dir>` flag (which sets `DIR`) but the `<dir>` specified didn't exist or was a file, it would silently not use it (confusing the user).

You have 2 choices:

1. Just remove `&& -d $DIR` from all of the lines you added it to. This will cause `pushd` to display an error msg if the directory is not valid.
2. Or do number 1 and add a check in `checkPrereqsAndInput` that the directory specified is valid, and display an error msg and return exit code 1 if not.

The fact that you didn't catch this problem makes me think you didn't test your code thoroughly enough before submitting the PR. Please test both the good path and the error cases. Thx.

Reply:
Yea, it should be reasonable. Thanks.